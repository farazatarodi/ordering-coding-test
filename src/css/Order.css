.order-container {
  width: 100%;
  display: flex;
  flex-direction: column;
  margin: 3rem 0;
}

.order-content {
  margin: 1rem;
  width: 100%;
  align-self: center;
  box-shadow: var(--shadow);
  border-radius: 1rem;
  border: 2px solid var(--color-neutral);
}

.order-header {
  width: 100%;
  display: grid;
  grid-template-columns: 10% auto 15% 15% 10%;
  padding: 2rem;
  background-color: var(--color-neutral-light);
  border-radius: var(--radius) var(--radius) 0 0;
}

.order-details {
  padding: 2rem;
  width: 100%;
  display: grid;
  grid-template-columns: 10% auto 15% 15% 10%;
  background-color: var(--color-neutral-lightest);
  border-top: 2px solid var(--color-neutral);
  border-radius: 0 0 var(--radius) var(--radius);
}

.order-content {
  margin: 1rem;
  width: 100%;
  align-self: center;
  box-shadow: var(--shadow);
  border-radius: 1rem;
  border: 2px solid var(--color-neutral);
}
