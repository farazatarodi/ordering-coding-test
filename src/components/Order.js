// component for individual order

// React imports
import PropTypes from 'prop-types';

// Router imports
import { Link } from 'react-router-dom';

// Redux imports
import { useSelector } from 'react-redux';

// component import
import Items from './Items';
import PlaceButton from './PlaceButton';

// css imports
import '../css/Order.css';

const Order = ({ id }) => {
  // get order from store by order ID
  const [order] = useSelector((state) =>
    state.ordersReducer.filter((e) => e.id === id)
  );

  // get customer from store by customer ID
  const [customer] = useSelector((state) =>
    state.customersReducer.customers.filter(
      (e) => e.id === order['customer-id']
    )
  );

  return (
    <div className="order-container">
      {/* link to main orders page */}
      <Link to="/orders" className="text text-display">
        {'< BACK'}
      </Link>
      <div className="order-content">
        <div className="order-header heading heading-2">
          <strong>Order ID</strong>
          <strong>Customer Name</strong>
          <strong>Customer ID</strong>
          <strong>Total</strong>
        </div>
        <div className="text text-display order-details">
          <div>{order && order.id}</div>
          <div>{customer && customer.name}</div>
          <div>{customer && customer.id}</div>
          <div>â‚¬{order && order.total}</div>
        </div>
      </div>

      {/* component that contains items */}
      <Items order={order} />

      {/* component that contains place order button and its logic */}
      <PlaceButton order={order} />
    </div>
  );
};

Order.propTypes = {
  id: PropTypes.string.isRequired,
};

export default Order;
