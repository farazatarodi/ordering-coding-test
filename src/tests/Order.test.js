import { render, screen } from '@testing-library/react';

import Order from '../components/Order';

import { Provider } from 'react-redux';

import configureStore from 'redux-mock-store';

import { BrowserRouter as Router } from 'react-router-dom';

import renderer from 'react-test-renderer';

const initialState = {
  ordersReducer: { orders: [] },
  customersReducer: { customers: [] },
  productsReducer: { products: [] },
};
const mockStore = configureStore();
const store = mockStore(initialState);

test('detail table is visibile in order page', () => {
  render(
    <Provider store={store}>
      <Router>
        <Order />
      </Router>
    </Provider>
  );

  expect(screen.getByText(/order id/i)).toBeTruthy();
  expect(screen.getByText(/â‚¬/i)).toBeTruthy();
});

test('empty item table is visible', () => {
  render(
    <Provider store={store}>
      <Router>
        <Order />
      </Router>
    </Provider>
  );

  expect(screen.getByText(/description/i)).toBeTruthy();
  expect(screen.queryByRole('link', { name: /delete/i })).toBeNull();
});

test('renders correclty', () => {
  const initialState = {
    customersReducer: { customers: [] },
    ordersReducer: { orders: [] },
    productsReducer: { products: [] },
  };
  const mockStore = configureStore();
  const store = mockStore(initialState);
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <Order />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
