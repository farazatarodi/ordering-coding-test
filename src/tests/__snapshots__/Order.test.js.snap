// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correclty 1`] = `
<div
  className="order-container"
>
  <a
    className="text text-display"
    href="/orders"
    onClick={[Function]}
  >
    &lt; BACK
  </a>
  <div
    className="order-content"
  >
    <div
      className="order-header heading heading-2"
    >
      <strong>
        Order ID
      </strong>
      <strong>
        Customer Name
      </strong>
      <strong>
        Customer ID
      </strong>
      <strong>
        Total
      </strong>
    </div>
    <div
      className="text text-display order-details"
    >
      <div />
      <div />
      <div />
      <div>
        â‚¬
      </div>
    </div>
  </div>
  <div
    className="items-container text text-display"
  >
    <div
      className="items-header heading heading-2"
    >
      <strong>
        ID
      </strong>
      <strong>
        Description
      </strong>
      <strong>
        Quantity
      </strong>
      <strong>
        Unit Price
      </strong>
      <strong>
        Total
      </strong>
    </div>
    <div
      className="card text text-display notification-card"
    >
      Nothing to show
    </div>
  </div>
  <button
    className="btn text text-display"
    onClick={[Function]}
    style={Object {}}
  >
    Order is empty
  </button>
</div>
`;
